python-avd/pyavd/_eos_designs/structured_config/metadata/utils.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Protocol
+
+from pyavd._errors import AristaAvdInvalidInputsError
+
+if TYPE_CHECKING:
+    from pyavd._eos_designs.schema import EosDesigns
+
+    from . import AvdStructuredConfigMetadataProtocol
+
+
+class UtilsMixin(Protocol):
+    """
+    Mixin Class with internal functions.
+
+    Class should only be used as Mixin to a AvdStructuredConfig class or other Mixins.
+    """
+
+    def get_resolved_validation_profile(self: AvdStructuredConfigMetadataProtocol, profile_name: str) -> EosDesigns.ValidationProfilesItem:
+        """
+        Return a fully resolved validation profile.
+
+        The method performs the following steps:
+        - Verifies that the requested validation profile exists.
+        - If a parent profile is defined, verifies that the parent exists.
+        - Resolves the profile by deep-inheriting from the parent profile.
+        - Removes the `parent_profile` attribute from the resolved profile
+        to prevent further inheritance processing.
+
+        Parameters:
+        ----------
```

Comment:
we use Google style for doc strings in AVD not numpy

(Also for numpy style)you should not have the final column)


```suggestion
        Parameters:
```

Reply:
updated